### PR TITLE
Demonstration sorting, Recent Progress and Overview Standards Window improvements

### PR DESCRIPTION
**Pull request by jmealo**
This PR implements the following:

1. Dashboard: Sort and filter standards demonstrations on the Student and Teacher dashboard as per requirements.
2. Standard Overview Window: Automatically expand the clicked demonstration
3. Standard Overview Window: Remove duplicate code from Student and Teacher dashboard using a common OverviewBody component 
4. Global: sort by the data demonstrated (ordering, dashboard, overview) or the date logged (Recent Progress) as appropriate
5. Student Dashboard: Wired up Recent Progress

### CODE REVIEW

File: php-classes/Slate/CBL/ContentAreasRequestHandler.php [hunk 1]
```diff
@@ -84,4 +86,56 @@ public static function handleCompetenciesRequest(ContentArea $ContentArea)
             'data' => $competencies
         ]);
     }
+    
+    public static function handleRecentProgressRequest($contentArea) {
+        $student = $_GET['student'];
+        $limit = isset($_GET['limit']) ? $_GET['limit'] : 10;
+        
+        if (!$student) {
+            if (!ctype_digit($student)) {
```
Comment (themightychris) [inline php-classes/Slate/CBL/ContentAreasRequestHandler.php]: @jmealo shouldn't these two conditions be OR'd together in the same if?

File: php-classes/Slate/CBL/DemonstrationSkill.php [hunk 2]
```diff
@@ -22,6 +22,9 @@ class DemonstrationSkill extends \ActiveRecord
             'type' => 'tinyint',
             'unsigned' => true
         ]
+        ,'Demonstrated' => [
+            'type' => 'date'
+        ]
```
Comment (themightychris) [inline php-classes/Slate/CBL/DemonstrationSkill.php]: @jmealo this field shouldn't be getting added to the DemonstrationSkill model, it's already stored in the related Demonstration model linked by DemonstrationID

File: sencha-workspace/packages/slate-cbl/src/view/student/DashboardController.js [hunk 3]
```diff
@@ -36,15 +36,26 @@ Ext.define('Slate.cbl.view.student.DashboardController', {
         var me = this,
             siteEnv = window.SiteEnvironment || {},
             student = siteEnv.cblStudent && Ext.create('Slate.cbl.model.Student', siteEnv.cblStudent)
-        studentDashboardCompetenciesList = dashboardView.el.down('#studentDashboardCompetenciesList'),
+            studentDashboardCompetenciesList = dashboardView.el.down('#studentDashboardCompetenciesList'),
+            /* HACK: what's the right way to get the recent progress... also why do we use down with an id above? */
+            studentDashboardRecentProgress = Ext.getElementById('studentDashboardRecentProgress'),
```
Comment (themightychris) [inline sencha-workspace/packages/slate-cbl/src/view/student/DashboardController.js]: @jmealo should just be Ext.get('student...Progress')

File: php-classes/Slate/CBL/ContentAreasRequestHandler.php (hunk 1 above)
Comment (jmealo) [inline php-classes/Slate/CBL/ContentAreasRequestHandler.php]: I think that I meant to use the currently logged in user as the student or something, I think the ``ctype_digit`` check would be sufficient.

File: php-classes/Slate/CBL/DemonstrationSkill.php (hunk 2 above)
Comment (jmealo) [inline php-classes/Slate/CBL/DemonstrationSkill.php]: Agreed.

File: sencha-workspace/packages/slate-cbl/src/view/student/DashboardController.js (hunk 3 above)
Comment (jmealo) [inline sencha-workspace/packages/slate-cbl/src/view/student/DashboardController.js]: Agreed. Will change when you're done making changes.

**Pull request merged**